db/db_impl/db_impl.cc

```diff
@@ -1801,6 +1810,19 @@ std::vector<Status> DBImpl::MultiGet(
       bytes_read += value->size();
       num_found++;
     }
+
+    if (read_options.deadline.count() &&
+        env_->NowMicros() >
+            static_cast<uint64_t>(read_options.deadline.count())) {
+      timedout = true;
+      break;
+    }
+  }
+
+  if (timedout) {
```

Comment:
Is `timedout` the equivalent to `keys_read < num_keys`?

Reply:
Yes. I suppose that check should suffice.